### PR TITLE
Feature/sequence number to projects

### PR DESCRIPTION
**Pull request by eman-ijaz7vals**

### CODE REVIEW

**Comment by eman-ijaz7vals** [comment]
Cycle 0 done

File: app/controllers/documents_controller.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class DocumentsController < ApplicationController
   before_action :authenticate_user!
-  load_resource :issue
-  load_resource :project
-  load_and_authorize_resource
+  load_resource :issue , :find_by => :sequence_num
+  load_resource :project , :find_by => :sequence_num
```
Comment (abubakarazeem) [inline app/controllers/documents_controller.rb]: new hash syntax

File: app/controllers/issues_controller.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ def filter
 
   # GET projects/:id/issues/new
   def new
-    @project = current_tenant.projects.find(params[:project_id])
+    @project = current_tenant.projects.find_by(sequence_num: params[:project_id])
```
Comment (abubakarazeem) [inline app/controllers/issues_controller.rb]: load through cancancan

File: app/controllers/documents_controller.rb [hunk 3]
```diff
@@ -25,10 +25,10 @@ def create
     @document.path = @document.document.url
     if params[:issue_id].present?
```
Comment (abubakarazeem) [inline app/controllers/documents_controller.rb]: use cancancan

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -23,7 +23,7 @@ def create
   end
 
   def show
-    @project = Project.find(params[:id])
+    # @project = Project.find(params[:id])
```
Comment (abubakarazeem) [inline app/controllers/projects_controller.rb]: remove this line

File: app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -23,7 +23,7 @@ def create
...
     @document = Document.new
     @issues = @project.issues
     @issue_types = IssueType.issue_types_for_projects(@project)
```
Comment (abubakarazeem) [inline app/controllers/projects_controller.rb]: better name for method

File: app/models/issue.rb [hunk 6]
```diff
@@ -1,5 +1,6 @@
 # Model Class
 class Issue < ActiveRecord::Base
+	sequenceid :company , :issues
```
Comment (abubakarazeem) [inline app/models/issue.rb]: indentation

File: app/models/project.rb [hunk 7]
```diff
@@ -1,4 +1,5 @@
 class Project < ActiveRecord::Base
+	sequenceid :company , :projects
```
Comment (abubakarazeem) [inline app/models/project.rb]: indentaion

File: app/views/issues/_issues.html.erb [hunk 8]
```diff
@@ -1,9 +1,9 @@
 <!-- Partial: Displays rows in a table -->
 <% issues.each do |issue| %>
   <tr>
-    <td><%= link_to issue.title, project_issue_path(issue.project_id, issue.id) %></td>
+    <td><%= link_to issue.title, project_issue_path(issue.project.sequence_num, issue.sequence_num) %></td>
     <% if request.path.include?('issues') %>
```
Comment (abubakarazeem) [inline app/views/issues/_issues.html.erb]: check for controller's action

File: app/views/issues/show.html.erb [hunk 9]
```diff
@@ -16,7 +16,7 @@
   </div>
 
   <% if can?(:update, @issue) %>
-    <%= link_to 'Edit', edit_project_issue_path(@issue.project_id, @issue.id), class: 'btn btn-secondary float-right mr-2' %>
+    <%= link_to 'Edit', edit_project_issue_path(@issue.project.sequence_num, @issue.sequence_num), class: 'btn btn-secondary float-right mr-2' %>
```
Comment (abubakarazeem) [inline app/views/issues/show.html.erb]: what if pass @issue.project only

File: app/views/layouts/_sidebar.html.erb [hunk 10]
```diff
@@ -15,14 +15,16 @@
...
-    <li>
-      <a href="<%= issue_types_path %>"><i class="material-icons">collections_bookmark</i>Issue Types</a>
-    </li>
-    <li>
-      <a href="<%= issue_states_path %>"><i class="material-icons">collections_bookmark</i>Issue States</a>
-    </li>
-    <li>
-      <a href="<%= issues_report_path %>"><i class="material-icons">view_module</i>&nbsp;Issues Report</a>
-    </li>
+    <% if can? :manage, :all %>
```
Comment (abubakarazeem) [inline app/views/layouts/_sidebar.html.erb]: if can? :manage, :all && true

File: app/views/layouts/_sidebar.html.erb [hunk 11]
```diff
@@ -15,14 +15,16 @@
...
+      <li>
+        <a href="<%= issue_types_path %>"><i class="material-icons">collections_bookmark</i>Issue Types</a>
```
Comment (abubakarazeem) [inline app/views/layouts/_sidebar.html.erb]: link_to(issue_types_path) d..

File: app/views/projects/index.html.erb [hunk 12]
```diff
@@ -1,5 +1,7 @@
 <h1 class="float-left">Projects</h1>
-<%= link_to 'New Project', new_project_path, class: 'btn btn-secondary float-right' %>
+<% if can? :manage, :all %>
```
Comment (abubakarazeem) [inline app/views/projects/index.html.erb]: if can? :create Project

File: app/views/projects/show.html.erb [hunk 13]
```diff
@@ -1,7 +1,9 @@
 <h1 class="float-left"><%= @project.title %></h1>
-<%= link_to 'Add Custom Issue Type', project_issue_types_path(@project),class: 'btn btn-secondary float-right mr-2' %>
-<%= link_to 'Edit Project', edit_project_path(@project),class: 'btn btn-secondary float-right mr-2' %>
-<%= link_to 'Delete Project', project_path(@project),class: 'btn btn-secondary float-right mr-2', method: 'delete', data: { confirm: 'Are you sure?' } %>
+<% if can? :manage, :all %>
```
Comment (abubakarazeem) [inline app/views/projects/show.html.erb]: check for every case

File: app/views/projects/show.html.erb [hunk 14]
```diff
@@ -1,7 +1,9 @@
...
+  <%= link_to 'Add Custom Issue Type', project_issue_types_path(@project),class: 'btn btn-secondary float-right mr-2' %>
+  <%= link_to 'Edit Project', edit_project_path(@project),class: 'btn btn-secondary float-right mr-2' %>
+  <%= link_to 'Delete Project', project_path(@project),class: 'btn btn-secondary float-right mr-2', method: 'delete', data: { confirm: 'Are you sure?' } %>
```
Comment (abubakarazeem) [inline app/views/projects/show.html.erb]: spacing

**Pull request merged**